ui/apps/platform/src/Containers/NetworkGraph/NetworkGraphPage.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Flex, FlexItem, PageSection } from '@patternfly/react-core';
+
+import useURLSearch from 'hooks/useURLSearch';
+import { fetchClustersAsArray } from 'services/ClustersService';
+import { Cluster } from 'types/cluster.proto';
+import ClusterSelect from './ClusterSelect';
+import NamespaceSelect from './NamespaceSelect';
+
+export type Namespace = {
+    metadata: {
+        id: string;
+        name: string;
+    };
+};
+
+function NetworkGraphPage() {
+    const [clusters, setClusters] = React.useState<Cluster[]>([]);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const selectedClusterId = (searchFilter.Cluster as string) || '';
+    const selectedNamespaces = (searchFilter.Namespace as string[]) || [];
+
+    React.useEffect(() => {
+        fetchClustersAsArray()
+            .then((data) => {
+                setClusters(data as Cluster[]);
+            })
+            .catch(() => {
+                // TODO
+            });
+    }, []);
+
+    function updateSelectedClusterId(selection) {
+        const newSearchFiliter = { ...searchFilter, Cluster: selection };
+        setSearchFilter(newSearchFiliter);
+    }
+
+    function updateSelectedNamespaces(selection) {
+        const newSearchFiliter = { ...searchFilter, Namespace: selection };
+        setSearchFilter(newSearchFiliter);
+    }
+
+    return (
+        <PageSection variant="light" isFilled id="policies-table-loading">
+            <h1>Network Graph</h1>
+            <Flex>
+                <FlexItem>
+                    <ClusterSelect
+                        clusters={clusters}
+                        selectedClusterId={selectedClusterId}
+                        setSelectedClusterId={updateSelectedClusterId}
+                    />
+                </FlexItem>
+                <FlexItem>
+                    <NamespaceSelect
```

Comment:
This Namespace Select component seems to do `data fetching` + `rendering` while the Cluster Select component only does `rendering` and does `data fetching` in this parent component. I think maybe we should try to keep the data fetching within this level and transform the data as necessary to be rendered in these Select components. I was thinking these Select components could even be a little more generic so that we can reuse them for all the other selects we'll need to have for this feature.